src/BaseActionWatcher.ts

```diff
@@ -28,26 +37,74 @@ export class BaseActionWatcher {
    * Start a polling loop
    */
   public async watch(isReplay: boolean = false) {
-    const startTime = new Date().getTime()
+    try {
+      this.running = true
+      const startTime = new Date().getTime()
```

Comment:
Change to Date.now(), its faster and more explicit https://jsperf.com/date-now-vs-new-date